src/modules/social-feed/comment/comment.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Comment } from "@prisma/client";
+import {
+  CreateCommentInput,
+  UpdateCommentInput,
+} from "src/core/dto/social-feed.dto.js";
+import { Service } from "typedi";
+
+import { prisma } from "../../../prisma/index.js";
+
+@Service()
+export default class CommentService {
+  async createComment(
+    commentInput: CreateCommentInput,
+    userId: number
+  ): Promise<Comment> {
+    try {
+      const comment = await prisma.comment.create({
+        data: {
+          userId: userId,
+          postId: commentInput.postId,
+          content: commentInput.content,
+        },
+      });
+      return comment;
+    } catch (error) {
+      throw new Error((error as Error).message);
+    }
+  }
+
+  async updateComment(
+    commentId: number,
+    updateInput: UpdateCommentInput
+  ): Promise<Comment> {
+    try {
+      const comment = await prisma.comment.update({
+        where: { id: commentId },
+        data: {
+          content: updateInput.content,
+        },
+      });
+      return comment;
+    } catch (error) {
+      throw new Error((error as Error).message);
+    }
+  }
+
+  async getCommentsByUserId(userId: number): Promise<Comment[]> {
```

Comment:
I would do `getCommentsByPostId` instead, or also? How would we use this as it is?

Reply:
Changed to getCommentsByPostId. Makes sense.